orbit_predictor/predictors/pass_iterators.py

```diff
@@ -201,84 +201,68 @@ def _find_los(self, tca):
 class SmartLocationPredictor(BaseLocationPredictor):
 
     def iter_passes(self):
-        current_date = self.start_date
-        while True:
-            valid, aos, tca, los, max_elevation = self._find_next_pass(current_date)
-            current_date = los
+        # Explore all values of t every 3 minutes
+        t_values = list(range(0, int((self.limit_date - self.start_date).total_seconds()), 180))
+        elev_values = []
+        for delta_seconds in t_values:
+            elev_values.append(self._elevation(delta_seconds))
 
-            if self.limit_date is not None and current_date > self.limit_date:
-                break
-            else:
-                if valid:
-                    yield PredictedPass(
-                        self.location,
-                        self.predictor.sate_id,
-                        max_elevation_deg=degrees(max_elevation),
-                        aos=aos,
-                        los=los,
-                        duration_s=(los - aos).total_seconds(),
-                        max_elevation_position=self.predictor.get_position(tca),
-                        max_elevation_date=tca,
-                     )
+        peaks_idx, _ = find_peaks(elev_values)
 
-    def _elevation_at(self, when_utc):
+        for peak_idx in peaks_idx:
+            if elev_values[peak_idx] <= 0:
```

Comment:
Should we leave some margin here? for example: 
`if elev_values[peak_idx] <= -0.1:`
I'm wondering the case in which 0.001 elevation passes are needed and the 0.0001 is inside the 3 minutes window we are using. Don't know what the proper margin would be.
If we add the margin, we have to add back the filter by elevation that is gone in this version (the `is_valid`).

Reply:
As the time resolution is 3 minutes, with the current `<= 0` we would be missing passes of at most 3 minutes duration. If we are worried about this, I would perhaps increase the time resolution, rather than a refinement in an extra step, so we don't have to put an arbitrary value here. What do you think?

Reply:
I think that for our use cases is ok as it is. Perhaps add a comment in the SmartLocationPredictor so other users are advised?

Reply:
Left a comment in both classes 👍🏽